Cython/Compiler/ModuleNode.py

```diff
@@ -2880,7 +3142,10 @@ def generate_module_creation_code(self, env, code):
                 env.method_table_cname,
                 doc,
                 env.module_cname))
-        code.putln("#else")
+        code.putln("#elif CYTHON_COMPILING_IN_LIMITED_API")
+        code.putln("PyState_AddModule(PyModule_Create(&%s), &%s);" % (
```

Comment:
Can `PyState_AddModule()` handle the case that `PyModule_Create()` fails here? (And, should it really handle it?)

Reply:
Yup! I thought it would be better to let `PyState_AddModule` handle the `NULL` here rather than having our own error handling: https://github.com/python/cpython/blob/master/Python/pystate.c#L698-L701

Since this exits I don't even need to check for the output of `PyState_AddModule`

Reply:
It doesn't seem to steal a reference, though, so this leaks a reference in the good case, right? Then I would suggest doing the usual create-test-use-decref dance. Much more obvious, much safer.

Reply:
Sounds good!

Reply:
Done! This should now be correct (unless I misused any of the temporary apis)